# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux24benh/95-inet-for-userspace.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

The in.h file need protection for some defines where the types are not known
to the user-space and pkt_sched.h missed the inclusion of the types.h
header which types it utilizes ...

 - Rene Rebe <rene@exactcode.de>

--- ./include/linux/in.h.orig	2003-08-28 22:11:50.000000000 +0200
+++ ./include/linux/in.h	2003-08-28 22:12:27.000000000 +0200
@@ -135,6 +135,8 @@
 	(sizeof(struct ip_msfilter) - sizeof(__u32) \
 	+ (numsrc) * sizeof(__u32))
 
+#ifdef __KERNEL__
+
 struct group_req
 {
 	__u32			gr_interface;	/* interface index */
@@ -161,6 +163,8 @@
 	(sizeof(struct group_filter) - sizeof(struct sockaddr_storage) \
 	+ (numsrc) * sizeof(struct sockaddr_storage))
 
+#endif
+
 struct in_pktinfo
 {
 	int		ipi_ifindex;
--- ./include/linux/pkt_sched.h.orig	2003-08-28 22:20:44.000000000 +0200
+++ ./include/linux/pkt_sched.h	2003-08-28 22:21:11.000000000 +0200
@@ -1,6 +1,8 @@
 #ifndef __LINUX_PKT_SCHED_H
 #define __LINUX_PKT_SCHED_H
 
+#include <linux/types.h>
+
 /* Logical priority bands not depending on specific packet scheduler.
    Every scheduler will map them to real traffic classes, if it has
    no more precise mechanism to classify packets.
